gradle/dependencies.gradle

```diff
@@ -152,7 +152,9 @@ versions += [
   // Also make sure the compression levels in org.apache.kafka.common.record.CompressionType are still valid
   zstd: "1.5.6-6",
   junitPlatform: "1.10.2",
-  hdrHistogram: "2.2.2"
+  hdrHistogram: "2.2.2",
+  log4j2: "2.24.1",
```

Comment:
https://github.com/apache/kafka/blob/2d896d9130f121e75ccba2d913bdffa358cf3867/build.gradle#L3354-L3375
Since `log4j-appender` is depending on `clients`, we will need to delete `log4j-appender` module if we want to entirely get rid of `log4j`.
https://issues.apache.org/jira/browse/KAFKA-17860

cc. @chia7712

Reply:
I have opened a PR for it #17588